pandas-stubs/core/series.pyi

```diff
@@ -1997,3 +2007,7 @@ class OffsetSeries(Series):
     def __radd__(self, other: Period) -> PeriodSeries: ...
     @overload
     def __radd__(self, other: BaseOffset) -> OffsetSeries: ...
+
+class IntervalSeries(Series, Generic[_OrderableT]):
+    @property
+    def array(self) -> IntervalArray: ...
```

Comment:
Why doesn't it work to add overloads instead of creating a helper class here? Not sure whether mypy finally accepts overload+property.

Reply:
> Why doesn't it work to add overloads instead of creating a helper class here? Not sure whether mypy finally accepts overload+property.

I couldn't see how to get even pyright to support `overload` with `property` .

Reply:
I thought at least pyright supports that - but I haven't tried it.

I think even if mypy+pyright support property+overload, we probably wouldn't want to use it as the first overload would need to be the more specific overload returning `IntervalArray` but we shouldn't return that for `Series[Any/Unknown]`.